### PR TITLE
Psh send host info to luoxiao

### PR DESCRIPTION
**Pull request by saying121**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -65,12 +64,27 @@ fn main() -> anyhow::Result<()> {
...
+    });
+
     if otlp_conf.enable() {
+        let rt = Arc::clone(&runtime);
         let export_conf: ExportConfig = otlp_conf.into();
-        let th = std::thread::spawn(|| {
-            tokio::runtime::Runtime::new()?.block_on(otlp::otlp_tasks(export_conf))
-        });
-        otlp_th = Some(th);
+        let hd = std::thread::spawn(move || rt.block_on(otlp::otlp_tasks(export_conf)));
```
Comment (BrytonLee) [inline src/main.rs:86]: Have you tested this branch?  I doubt it wouldn't work as `rt` is not initialized before the new created thread `hd`,  my understanding is that tokio runtime manages a bunch of threads, and set thread local variable to reference to tokio runtime, a user created thread may not be managed by tokio and thus cannot simply "move" a tokio runtime into it.

File: src/services/host_info.rs [hunk 2]
```diff
@@ -0,0 +1,104 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use sysinfo::System;
```
Comment (BrytonLee) [inline src/services/host_info.rs:3]: I am not sure if `crate/psh-system` can provide the host information you need, but I think these information should be provided in `crate/psh-system` and gRPC service simply call `crate/psh-system` public functions to encapsulate data for needs.

**Review by BrytonLee — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 1 above)
Comment (saying121) [reply src/main.rs:86]: It works fine. In the tokio doc [share runtime](https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#sharing) there is a way to share runtime and run on another thread and prevent the runtime from shutting down.

File: src/services/host_info.rs (hunk 2 above)
Comment (saying121) [reply src/services/host_info.rs:3]: I will have a look

File: src/main.rs (hunk 1 above)
Comment (BrytonLee) [reply src/main.rs:86]: Well, it is true you can share a tokio runtime between threads by using `Arc`. The situation becomes if it is necessary to share this tokio runtime with a new created thread.  Will `otlp::otlp_tasks(export_confg)` run inside this new created thread or it is managed by tokio runtime, no matter this new thread? 

You can do an experiment, comment `PshEngineBuilder` code to see if psh will exit immediately, if so, then create a new thread here is not necessary.

Comment (saying121) [reply src/main.rs:86]: Psh not exit immediately.

**Pull request closed without merging**